vcd/resource_vcd_vapp_network_test.go

```diff
@@ -0,0 +1,160 @@
+		"endAddress":       "192.168.1.20",
+		"vappName":         vappNameForNetworkTest,
+	}
+
+	configText := templateFill(testAccCheckVappNetwork_basic, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVappNetworkDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVappNetworkExists("vcd_vapp_network."+resourceName),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_network."+resourceName, "gateway", gateway),
+					resource.TestCheckResourceAttr(
+						"vcd_vapp_network."+resourceName, "dns1", dns1),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckVappNetworkExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no vapp network ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*VCDClient)
+
+		found, err := isVappNetworkFound(conn, rs)
+		if err != nil {
+			return err
+		}
+
+		if !found {
+			return fmt.Errorf("vApp network was not found")
+		}
+
+		return nil
+	}
+}
+
+// TODO: In future this can be improved to check if network delete only,
+// when test suite will create vApp which can be reused
+func testAccCheckVappNetworkDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*VCDClient)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "vcd_vapp" {
+			continue
+		}
+
+		_, err := isVappNetworkFound(conn, rs)
+		if err != nil && !strings.Contains(err.Error(), "can't find vApp:") {
+			return fmt.Errorf("vapp %s still exist and error: %#v", itemName, err)
+		}
+	}
+
+	return nil
+}
+
+func isVappNetworkFound(conn *VCDClient, rs *terraform.ResourceState) (bool, error) {
+	_, vdc, err := conn.GetOrgAndVdc(testConfig.VCD.Org, testConfig.VCD.Vdc)
+	if err != nil {
+		return false, fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	vapp, err := vdc.FindVAppByName(vappNameForNetworkTest)
+	if err != nil {
+		return false, fmt.Errorf("error retrieving vApp: %s, %#v", rs.Primary.ID, err)
+	}
+
+	networkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return false, fmt.Errorf("error retrieving network config from vApp: %#v", err)
+	}
+
+	var found bool
+	for _, vappNetworkConfig := range networkConfig.NetworkConfig {
+		if vappNetworkConfig.NetworkName == newVappNetworkName && vappNetworkConfig.Configuration.IPScopes.IPScope.DNSSuffix == dnsSuffix {
+			found = true
+		}
+	}
+
+	return found, nil
+}
+
+const testAccCheckVappNetwork_basic = `
+
+resource "vcd_vapp" "{{.vappName}}" {
+  name = "{{.vappName}}"
+  org  = "{{.Org}}"
+  vdc  = "{{.Vdc}}"
+}
+
+resource "vcd_vapp_network" "{{.resourceName}}" {
+  org          = "{{.Org}}"
```

Comment:
This snippet is not formatted correctly. The properties should be aligned.

Reply:
Fixed